### PR TITLE
fix compilation without deprecated OpenSSL APIs

### PR DESCRIPTION
**Pull request by neheb**
Initialization is deprecated.

### CODE REVIEW

File: lib/crypt.cpp [hunk 1]
```diff
@@ -718,8 +719,10 @@ char *check_validity(
     int rbytes;
     unsigned char md5_md[MD5_DIGEST_LENGTH],  rbuf[2048];
 
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(HAVE_LIBRESSL)
```
Comment (AenBleidd) [inline lib/crypt.cpp:722]: Please put comment with string representation of version for better reading

File: lib/crypt.cpp [hunk 2]
```diff
@@ -770,7 +773,9 @@ int cert_verify_file(
         fflush(stdout);
         return false;
     }
+ #if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(HAVE_LIBRESSL)
```
Comment (AenBleidd) [inline lib/crypt.cpp:776]: Please put comment with string representation of version for better reading

File: lib/crypt.cpp [hunk 3]
```diff
@@ -42,6 +42,7 @@
 #include <openssl/engine.h>
 #include <openssl/err.h>
 #include <openssl/rsa.h>
+#include <openssl/bn.h>
```
[the remark this answers is not in the document]
Comment (neheb) [reply lib/crypt.cpp:45]: for BN_ function usage. This header is normally included with openssl/ssl.h. Without deprecated APIs it is not.

File: lib/crypt.cpp (hunk 1 above)
Comment (neheb) [reply lib/crypt.cpp:723]: Done

File: lib/crypt.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (AenBleidd) [reply lib/crypt.cpp:45]: According to this https://github.com/openssl/openssl/commit/98186eb4e4aef6262ed6b0f499348defa2c26893#diff-65f95c90a6c44264a23d1e37890c4e20R74-R75
this files should be included when `OPENSSL_VERSION_NUMBER < 0x10100000L`
I suggest you to add the same check here as in another files in this PR:
`#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(HAVE_LIBRESSL)`

[the remark this answers is not in the document]
Comment (AenBleidd) [reply lib/crypt.cpp:45]: But the logic looks strange. I don't know exactly why this file is not included in later versions of OpenSSL, so maybe these BN_* functions usages should be completely rewritten to support newest OpenSSL

[the remark this answers is not in the document]
Comment (neheb) [reply lib/crypt.cpp:45]: No idea what the point of doing this is. It's just a header file.

It makes sense for ssl.h as some deprecated functions in SSL.h rely on it.

**Comment by neheb** [comment]
```
freetype.lib(sfnt.c.obj) : error LNK2019: unresolved external symbol BrotliDecoderDecompress referenced in function woff2_decompress [C:\projects\boinc\win_build\slide_show_vs2019.vcxproj]
```

Doesn't seem related.

**Comment by AenBleidd** [comment]
> ```
> freetype.lib(sfnt.c.obj) : error LNK2019: unresolved external symbol BrotliDecoderDecompress referenced in function woff2_decompress [C:\projects\boinc\win_build\slide_show_vs2019.vcxproj]
> ```
> 
> Doesn't seem related.

It's true. Fix for this issue in the another PR.
I want to  investigate more about deprecated functions and changes you made before merging this.

**Comment by neheb** [comment]
Just compile with -DOPENSSL_API_COMPAT=0x10100000L and -Werror=implicit-function-declaration to simulate.

**Comment by ChristianBeer** [comment]
Close and reopen to trigger new CI build.

**Comment by ChristianBeer** [comment]
Thanks for contributing.

**Pull request merged**